examples/cpp/dds/TypeLookupService/TypeLookupPublisher.cpp

```diff
@@ -52,13 +53,21 @@ bool TypeLookupPublisher::init()
         return false;
     }
 
-    types::DynamicType_ptr dyn_type = xmlparser::XMLProfileManager::getDynamicTypeByName("TypeLookup")->build();
-    TypeSupport m_type(new types::DynamicPubSubType(dyn_type));
-    m_Hello = types::DynamicDataFactory::get_instance()->create_data(dyn_type);
+    DynamicTypeBuilder* dyn_build {nullptr};
+    if (xmlparser::XMLP_ret::XML_OK !=
+            xmlparser::XMLProfileManager::getDynamicTypeByName(dyn_build, "TypeLookup"))
```

Comment:
The API is inconsistent. In the DynamicHelloWorldExample this method return a DynamicType and not a DynamicTypeBuilder. I have looked into the XMLProfileManager API and the correct one is the DynamicType.

Reply:
Obsolete